source/integrations/fastapi-integration.txt

```diff
@@ -0,0 +1,540 @@
+        # This will be aliased to ``_id`` when sent to MongoDB,
+        # but provided as ``id`` in the API requests and responses.
+        id: Optional[PyObjectId] = Field(alias="_id", default=None)
+        name: str = Field(...)
+        email: EmailStr = Field(...)
+        course: str = Field(...)
+        gpa: float = Field(..., le=4.0)
+        model_config = ConfigDict(
+            populate_by_name=True,
+            arbitrary_types_allowed=True,
+            json_schema_extra={
+                "example": {
+                    "name": "Jane Doe",
+                    "email": "jdoe@example.com",
+                    "course": "Experiments, Science, and Fashion in Nanophotonics",
+                    "gpa": 3.0,
+                }
+            },
+        )
+
+UpdateStudentModel Class
+`````````````````````````
+
+The ``UpdateStudentModel`` has some key differences from the ``StudentModel``:
+
+-   It does not have an ``id`` attribute, as this cannot be modified
+-   All fields are optional, so you can supply only the fields you want to update
+-   The ``mongo_config`` variables includes ``json_encoders={ObjectId: str}`` 
+
+FastAPI encodes and decodes data as JSON strings, which do not support
+all the data types that MongoDB's BSON data type can store. BSON has
+support for more non-JSON-native data types, including ``ObjectId``
+which is used for the default UUID attribute, ``_id``. Because of this,
+you must convert ``ObjectId`` objects to strings before storing them in
+the ``_id`` field. The ``json_encoders`` setting instructs Pydantic to do this.
+
+For more information about how BSON compares to JSON, see the `JSON
+and BSON <https://www.mongodb.com/json-and-bson>`__ MongoDB article. 
+
+You can see the ``UpdateStudentModel`` definition in the following code in the ``app.py`` file:
+
+.. code-block:: python
+
+   class UpdateStudentModel(BaseModel):
+       """
+       A set of optional updates to be made to a document in the database.
+       """
+
+       name: Optional[str] = None
+       email: Optional[EmailStr] = None
+       course: Optional[str] = None
+       gpa: Optional[float] = None
+       model_config = ConfigDict(
+           arbitrary_types_allowed=True,
+           json_encoders={ObjectId: str},
+           json_schema_extra={
+               "example": {
+                   "name": "Jane Doe",
+                   "email": "jdoe@example.com",
+                   "course": "Experiments, Science, and Fashion in anophotonics",
+                   "gpa": 3.0,
+               }
+           },
+       )
+
+StudentCollection Class
+````````````````````````
+
+The ``StudentCollection`` class is defined to encapsulate a list of
+``StudentModel`` instances. The purpose of this endpoint is to provide some protection from JSON Hijacking. For more information, you can read this article about `JSON Hijacking
+<https://haacked.com/archive/2009/06/25/json-hijacking.aspx/>`__. 
```

Comment:
S: define link target more clearly
eg "Read the "xyz" article on the Haacked website.